pkg/operator/ceph/cluster/osd/update.go

```diff
@@ -169,6 +170,35 @@ func (c *updateConfig) updateExistingOSDs(errs *provisionErrors) {
 			continue
 		}
 
+		// handle cephx key rotation for the OSD
```

Comment:
This block won't be executed if the PGs are not clean. Is that the desired behavior?

Reply:
From huddle chat:

This wasn't necessarily intended, but this is the simplest place in the code to add key rotation. As written, users should be able to use the CephCluster settings to force updates if they need rotation while PGs are unhealthy. For now, since a workaround exists, let's assume the behavior is okay, and we can revisit later if we see a need. Thanks for raising this